apps/verticalhome/js/pin_app_navigation.js

```diff
@@ -0,0 +1,294 @@
+'use strict';
+/* global ApplicationSource */
+/* global BookmarkSource */
+/* global CollectionSource */
+/* global configurator */
+/* global dispatchEvent */
+/* global GaiaGrid */
+
+(function(exports) {
+
+
+  /**
+   * This is PinAppNavigation's constructor.
+   * @param {Object} list [DOM container that contains pin elements]
+   */
+  function PinAppNavigation(list) {
+    if (!list && typeof list !== 'object') {
+      console.log('List of pin apps must be an object');
+      return;
```

Comment:
constructors typically should not return values. The appropriate thing to do here is to throw an exception.  Or just assume that all the call sites will get do the right thing and pass an object and remove the error checking.